### DISCUSSION TITLE
Questions about support sets for Wasserstein barycenters

### DISCUSSION THREAD

**Discussion by sandeshAdhikary** [PythonOT/POT#367] [Q&A] [answered]
Hello! Thank you all for all the great work on this excellent package!

I have a few questions regarding Wasserstein barycenter algorithms in POT. I am interested in computing Wasserstein barycenters of multiple empirical distributions such that (1) each distribution is defined over a different support set of samples and (2) the ground metric is arbitrary.  Here are my questions:

1. **Are there any solvers in POT for free-support barycenters with respect to arbitrary ground metrics?** The only free-support algorithm seems to be `ot.lp.free_support_barycenter` based on Algorithm 1 from this [paper](https://arxiv.org/pdf/1310.4375.pdf). But as far as I can tell, it does not allow for arbitrary ground metrics.

2. **Do all fixed-support Wasserstein barycenter solvers in POT assume that all input distributions (and the barycenter itself) are defined over the same support set?** For instance, `ot.barycenter` only takes a single cost _matrix_ as input, so I’m guessing this matrix should encode pairwise distances between samples in a single fixed support set. All distributions are then defined as histograms over the same support?

**Comment by rflamary** [comment, marked as answer]
1. There is for the moment no "general ground metrics" free_support_barycenter because the algorithm implemented is a block coordinate descent that requires a closed form solution for the update of the position which is not available for any losses. For general loss you can indeed implement a gradient descent form a sum of losses  returned by `ot.emd2` when using differentiable loss matrices.

2. The `ot.barycenter`function supposes that the support for all the distributions in the barycenter is the same (because a ground cost matrix is necessary and the j index in M_i,j index the positions in the distributions) but the barycenter itself does not need to be the same support as the distributions (the ground metric need not be square and the i index can correspond to different positions on the barycenter).

**Reply by sandeshAdhikary** [reply]
Thank you! That clears things up
